util/slicesx/slicesx.go

```diff
@@ -77,3 +77,15 @@ func EqualSameNil[S ~[]E, E comparable](s1, s2 S) bool {
 	}
 	return true
 }
+
+// ReplaceBetween replaces the elements between start and end exclusive with the
+// given replacement. It returns a new slice with the replacement applied.
+func ReplaceBetween[S ~[]T, T any](s S, start, end int, replacement []T) S {
```

Comment:
I'd rather not add this to slicesx. It seems pretty specialized and I don't foresee having other callers.

If we later did, then we could move it here.